skypy/utils/random.py

```diff
@@ -66,5 +66,39 @@ def _schechter_cdf(x, x_min, x_max, alpha):
     b = special.gammaincc(alpha + 1, x)
     c = special.gammaincc(alpha + 1, x_min)
     d = special.gammaincc(alpha + 1, x_max)
-
     return (a - b) / (c - d)
+
+
+def conditional_prob_shmr(x_min, x_max, size=None):
```

Comment:
Sometimes, naming functions can be tricky. In SkyPy we tend to avoid abbreviations (although in your case it can be very cumbersome). I would suggest renaming the function: `conditional_probability_shmr`.
Q: is this function likely to be used within another context beyond SHMR? If so, maybe we could drop `shmr` from the name.
These are only suggestions.